presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClientModule.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.oracle;
+
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.DriverConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcClient;
+import io.prestosql.plugin.jdbc.credential.CredentialProvider;
+import oracle.jdbc.OracleDriver;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class OracleClientModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(JdbcClient.class).to(OracleClient.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(BaseJdbcConfig.class);
+        configBinder(binder).bindConfig(OracleConfig.class);
+    }
+
+    @Provides
+    @Singleton
+    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig)
+            throws SQLException
+    {
+        Properties connectionProperties = new Properties();
+        connectionProperties.setProperty("defaultRowPrefetch", String.valueOf(oracleConfig.getDefaultRowPrefetch()));
```

Comment:
This seems to be [legacy](https://docs.oracle.com/en/database/oracle/oracle-database/19/jajdb/oracle/jdbc/OracleConnection.html#setDefaultRowPrefetch_int_):

> Do not mix the JDBC 2.0 fetch size API and the Oracle row-prefetching API in your application. You can use one or the other but not both.

Instead, we should call the standard `setFetchSize()` on `Statement` in `OracleClient.getPreparedStatement()` like we do for PostgreSQL.

Let's rename the config to `fetch-size` to better match the behavior ("default" applies to the behavior of this applying as a default for all created connections ... but to the end user, it is simply setting the fetch size).

Reply:
config renamed and added a getPreparedStatement..
but how I can get the config on that method? this config must be saved on contructor?